### PR TITLE
Fixes #1675

### LINKED ISSUE #1675: In-course links in Notify content disable scrolling
If you include a hyperlink to another part of the course inside a Notify popup, the link works but scrolling is subsequently completely disabled.

This is presumably down to notifyView disabling scrolling when a Notify popup gets opened - but is not being given the chance to re-enable it again because clicking the link will cause Adapt to route to the new location without calling `closeNotify`.

Looking at the code I suspect this will also have an impact on accessibility, though I haven't tested that.

It seems reasonable to me that someone would want to add a link to another part of the course to a Notify popup - in this particular case it was to direct the user to a part of the course they needed to revisit because they'd answered a question incorrectly - so it would be great to get this working.

### PR DESCRIPTION
**Pull request by danielstorey**

### CODE REVIEW

**Comment by moloko** [comment]
Works for me... tested in Firefox

File: src/core/js/views/notifyView.js [hunk 1]
```diff
@@ -91,7 +92,13 @@ define([
             event.preventDefault();
             //tab index preservation, notify must close before subsequent callback is triggered
             this.closeNotify();
-            Adapt.trigger("notify:cancelled");
+            Adapt.trigger('notify:cancelled');
+        },
+
+        onLinkClicked: function(e) {
```
Comment (moloko) [inline src/core/js/views/notifyView.js]: I think this could use an explanatory comment, maybe including a link to #1675

**Comment by danielstorey** [comment]
I think a lot of code could do with explanatory comments ;)

**Comment by danielstorey** [comment]
Removed link event handler. Instead call closeNotify on remove which will ensure the necessary happens when the route changes. Adding a listener to page:scrollTo ensures that stuff happens when navigating to an element on the same page.

File: src/core/js/views/notifyView.js [hunk 2]
```diff
@@ -19,7 +19,8 @@ define([
         },
 
         setupEventListeners: function() {
-            this.listenTo(Adapt, 'remove', this.remove);
+            this.listenTo(Adapt, 'remove', this.closeNotify);
```
Comment (moloko) [inline src/core/js/views/notifyView.js]: Now that there's quite a few of these, would you mind changing this to the object style i.e.
```js
this.listenTo(Adapt, {
    'remove', this.closeNotify,
    'page:scrollTo': this.closeNotify,
    'device:resize': this.resetNotifySize,
    'accessibility:toggle': this.onAccessibilityToggle
});
```

Comment (danielstorey) [inline src/core/js/views/notifyView.js]: Good call, I've done as you suggested Matt, as well as condensing remove and page:scrollTo to a single key

**Pull request merged**